client/lib/promote-post/index.ts

```diff
@@ -71,6 +72,34 @@ export async function loadDSPWidgetJS(): Promise< void > {
 	await import( './string' );
 }
 
+const ANDROID_VERSION_HIDE_CAMPAIGNS_BUTTON = 22.9;
+
+const shouldHideGoToCampaignButton = () => {
+	// Android versions higher or equal than 22.9 should hide the button
+	// Trycatching it since user-agents can be changed to send custom data
+	try {
+		const userAgent = navigator.userAgent.toLowerCase();
```

Comment:
It would be a good idea to create a new utility method inside the `client/lib/mobile-app/index.js` library, to read the device info. It it could return something like this `{ device: 'android', version: '29.1' }`, or we can have two methods for getting that info.

in the future, we may need to adjust other parts of the dashboard, and having that code living there would be good.